### PR TITLE
Sprint 3

### PR DESCRIPTION
**Pull request by AdrianChojnacki**

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { Controller } from "react-hook-form";
+import { TextField } from "@mui/material";
+import { DesktopDatePicker } from "@mui/x-date-pickers";
+import DatePickerCSS from "./DatePicker.module.css";
```
Comment (mosilver) [inline src/components/DatePicker/DatePicker.tsx]: I like order with imports - the css could be on lower position

File: src/components/DeleteButton/DeleteButton.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { AiFillDelete } from "react-icons/ai";
+import { ListReloadSetTrueContext } from "../../providers";
+import DeleteButtonCSS from "./DeleteButton.module.css";
+
+const DeleteButton = ({ id }: { id: number }) => {
+  const setReloadTrue = useContext(ListReloadSetTrueContext);
+
+  const url = `${process.env.REACT_APP_API_URL}/${id}`;
+
+  const handleClick = () => {
```
Comment (mosilver) [inline src/components/DeleteButton/DeleteButton.tsx]: I don't like the axios logic in this place - the delete button should be a dumb component and the action should be defined in the parent

File: src/components/DeleteButton/DeleteButton.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    axios.delete(url).catch((err) => {
+      console.log(err);
+    });
+
+    setReloadTrue();
```
Comment (mosilver) [inline src/components/DeleteButton/DeleteButton.tsx]: I think you should use useEffect hook (in parent) to listen if the list has changed

File: src/components/DeleteButton/DeleteButton.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  };
+
+  return (
+    <span onClick={handleClick} className={DeleteButtonCSS.btn}>
```
Comment (mosilver) [inline src/components/DeleteButton/DeleteButton.tsx]: why it's a span? you haven't accessibility support there

File: src/components/DeleteButton/DeleteButton.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { ListReloadSetTrueContext } from "../../providers";
+import DeleteButtonCSS from "./DeleteButton.module.css";
+
+const DeleteButton = ({ id }: { id: number }) => {
+  const setReloadTrue = useContext(ListReloadSetTrueContext);
+
+  const url = `${process.env.REACT_APP_API_URL}/${id}`;
+
+  const handleClick = () => {
+    axios.delete(url).catch((err) => {
```
Comment (mosilver) [inline src/components/DeleteButton/DeleteButton.tsx]: before delete you should show a prompt to confirm

File: src/components/Header/Header.tsx [hunk 6]
```diff
@@ -3,8 +3,8 @@ import { Stack, Button } from "@mui/material";
 import HeaderCSS from "./Header.module.css";
 
 const texts = {
-  invoices: "INVOICES",
-  create: "CREATE",
+  invoices: "INVOICES LIST",
+  create: "CREATE INVOICE",
```
Comment (mosilver) [inline src/components/Header/Header.tsx]: how about translations? (check the npm dependency list in upskill doc)

File: src/components/InvoiceCreate/InvoiceCreate.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+          <DatePicker
+            name="until"
+            control={control}
+            label="Payment date"
+            defaultDate={new Date()}
+            disabledPast
+          />
+          {errors.noPaymentDate && <ErrorMessage text="Pick a date" />}
+          {errors.paymentDateBeforeCreationDate && (
+            <ErrorMessage text="Payment date cannot be before creation date" />
```
Comment (mosilver) [inline src/components/InvoiceCreate/InvoiceCreate.tsx]: I would keep all static labels in literal & think about translations (json)

File: src/components/hoc/withFormSubmit/withFormSubmit.tsx [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+    const formValidation: IFormValidation = (created, until, amount) => {
+      let noCreationDate = false;
+      let noPaymentDate = false;
+      let paymentDateBeforeCreationDate = false;
+      let noAmount = false;
+
+      if (!created) noCreationDate = true;
+      if (!until) noPaymentDate = true;
+      if (created && until && created > until)
```
Comment (mosilver) [inline src/components/hoc/withFormSubmit/withFormSubmit.tsx]: this statements should have brackets {}

File: src/components/hoc/withFormSubmit/withFormSubmit.tsx [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+          })
+          .then(() => {
+            setSubmitReload(false);
+            showPopup();
+          })
+          .catch((err) => {
+            console.log(err);
+          });
+      }
+      setReloadTrue();
```
Comment (mosilver) [inline src/components/hoc/withFormSubmit/withFormSubmit.tsx]: strange approach

File: src/components/InvoiceDetails/InvoiceDetails.types.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+export interface IInvoiceDetailsProps {
+  invoiceData: {
+    id: number;
+    created: string;
```
Comment (mosilver) [inline src/components/InvoiceDetails/InvoiceDetails.types.ts:4]: I think it could be Date ?

File: src/components/DatePicker/DatePicker.tsx (hunk 1 above)
Comment (AdrianChojnacki) [reply src/components/DatePicker/DatePicker.tsx]: Done

File: src/components/DeleteButton/DeleteButton.tsx (hunk 2 above)
Comment (AdrianChojnacki) [reply src/components/DeleteButton/DeleteButton.tsx]: Done

File: src/components/DeleteButton/DeleteButton.tsx (hunk 3 above)
Comment (AdrianChojnacki) [reply src/components/DeleteButton/DeleteButton.tsx]: Done

File: src/components/DeleteButton/DeleteButton.tsx (hunk 4 above)
Comment (AdrianChojnacki) [reply src/components/DeleteButton/DeleteButton.tsx]: Done

File: src/components/DeleteButton/DeleteButton.tsx (hunk 5 above)
Comment (AdrianChojnacki) [reply src/components/DeleteButton/DeleteButton.tsx]: Done

File: src/components/hoc/withFormSubmit/withFormSubmit.tsx (hunk 8 above)
Comment (AdrianChojnacki) [reply src/components/hoc/withFormSubmit/withFormSubmit.tsx]: Done

File: src/components/hoc/withFormSubmit/withFormSubmit.tsx (hunk 9 above)
Comment (AdrianChojnacki) [reply src/components/hoc/withFormSubmit/withFormSubmit.tsx]: Done

File: src/components/InvoiceDetails/InvoiceDetails.types.ts (hunk 10 above)
Comment (AdrianChojnacki) [reply src/components/InvoiceDetails/InvoiceDetails.types.ts:4]: It is already formatted date. I format it in withFormSubmit.tsx in the moment of submitting the form. Then I show it already formatted in Details and Edit pages.

File: src/components/Header/Header.tsx (hunk 6 above)
Comment (AdrianChojnacki) [reply src/components/Header/Header.tsx]: Done

File: src/components/InvoiceCreate/InvoiceCreate.tsx (hunk 7 above)
Comment (AdrianChojnacki) [reply src/components/InvoiceCreate/InvoiceCreate.tsx]: Done

**Review by mosilver — APPROVED** [review]

**Review by mosilver — APPROVED** [review]

**Pull request merged**